lnbits/wallets/breez.py

```diff
@@ -0,0 +1,161 @@
+            )
+
+else:
+    import asyncio
+    from typing import AsyncGenerator, Optional
+
+    from loguru import logger
+
+    from lnbits import bolt11 as lnbits_bolt11
+    from lnbits.settings import settings
+
+    from .base import (
+        InvoiceResponse,
+        PaymentResponse,
+        PaymentStatus,
+        StatusResponse,
+        Wallet,
+    )
+
+    breez_event_queue: asyncio.Queue = asyncio.Queue()
+
+    class SDKListener(
+        breez_sdk.EventListener  # pyright: ignore[reportUnboundVariable]
+    ):
+        def on_event(self, event):
+            logger.debug(event)
+            breez_event_queue.put_nowait(event)
+
+    class BreezSdkWallet(Wallet):  # type: ignore[no-redef]
+        def __init__(self):
+            assert settings.breez_greenlight_seed, "missing Greenlight seed"
+            self.breez_greenlight_seed = breez_sdk.mnemonic_to_seed(
+                settings.breez_greenlight_seed
+            )
+            assert settings.breez_api_key, "missing Breez api key"
+            self.breez_api_key = settings.breez_api_key
+            assert (
+                settings.breez_greenlight_invite_code
+            ), "missing Greenlight invite code"
+            self.breez_greenlight_invite_code = settings.breez_greenlight_invite_code
+
+            self.config = breez_sdk.default_config(
+                breez_sdk.EnvironmentType.PRODUCTION,
+                self.breez_api_key,
+                breez_sdk.NodeConfig.GREENLIGHT(
+                    config=breez_sdk.GreenlightNodeConfig(
+                        partner_credentials=None,
+                        invite_code=self.breez_greenlight_invite_code,
+                    )
+                ),
+            )
+            self.sdk_services = breez_sdk.connect(
+                self.config, self.breez_greenlight_seed, SDKListener()
+            )
+
+        async def cleanup(self):
+            self.sdk_services.disconnect()
+
+        async def status(self) -> StatusResponse:
+            try:
+                node_info: breez_sdk.NodeState = self.sdk_services.node_info()
+            except Exception as exc:
+                return StatusResponse(f"Failed to connect to breez, got: '{exc}...'", 0)
+
+            return StatusResponse(None, int(node_info.max_payable_msat))
+
+        async def create_invoice(
+            self,
+            amount: int,
+            memo: Optional[str] = None,
+            description_hash: Optional[bytes] = None,
+            unhashed_description: Optional[bytes] = None,
+            **kwargs,
+        ) -> InvoiceResponse:
+            # if description_hash or unhashed_description:
+            #     raise Unsupported("description_hash and unhashed_description")
+
+            breez_invoice: breez_sdk.ReceivePaymentResponse = (
+                self.sdk_services.receive_payment(
+                    breez_sdk.ReceivePaymentRequest(
+                        amount,
```

Comment:
breez expects msats but in UI its sats so the amount will be off by 1000

Reply:
Ah, that explains the issue I saw when I was testing it!

Reply:
same problem on the wallet balance, just the other way around - breez returns msats so the balance in lnbits is off by x1000

or maybe i've just discovered a bitcoin faucet :D

Reply:
Fixed in 9c54b11241f89527342ef5d68a419c40a18cad7c

Reply:
> same problem on the wallet balance, just the other way around - breez returns msats so the balance in lnbits is off by x1000

I don't see this issue. When I set 2501 sats to LNbits, it showed me 2501 sats.